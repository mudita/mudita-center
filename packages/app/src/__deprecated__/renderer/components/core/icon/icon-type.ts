/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

export enum IconType {
  Arrow,
  ArrowLongLeft,
  ArrowLongRight,
  AttachContact,
  BackupFolder,
  Battery,
  BorderCheckIcon,
  Calendar,
  CalendarIcon,
  ChargingBattery,
  Check,
  CheckCircle,
  CheckIndeterminate,
  Cloud,
  Close,
  CloseWhite,
  Delete,
  FilesManager,
  HighBattery,
  HighRange,
  LowBattery,
  LowRange,
  Magnifier,
  MenuFilesManager,
  MenuHelp,
  MenuContacts,
  MenuMeditation,
  MenuMusic,
  MenuNews,
  MenuOverview,
  MenuPhone,
  MenuSettings,
  MenuTethering,
  MenuTools,
  MenuRange,
  MenuBattery,
  Message,
  ExternalLink,
  MuditaLogo,
  MuditaLogoWithText,
  MuditaLogoVertical,
  MediumBattery,
  MediumRange,
  NoBattery,
  NoRange,
  Refresh,
  Send,
  Sim,
  Support,
  Template,
  Calls,
  Connection,
  Contact,
  NewContact,
  Dial,
  Notes,
  Notifications,
  Playlist,
  PlusSign,
  Reload,
  Upload,
  UploadDark,
  VeryHighBattery,
  VeryHighRange,
  VeryLowBattery,
  VeryLowRange,
  VoiceRecorder,
  Info,
  Download,
  DownloadWhite,
  Pure,
  Fail,
  FailRed,
  More,
  Blocked,
  Forward,
  Edit,
  Favourites,
  Ice,
  ArrowDown,
  ArrowDownWhite,
  DeleteBig,
  Attachment,
  Remove,
  SendButton,
  Backup,
  Tooltip,
  TetheringStatus,
  Apple,
  Google,
  SynchronizeContacts,
  Templates,
  MissedCall,
  ConferenceCall,
  IncomingCall,
  OutgoingCall,
  MusicGrey,
  VoiceRecorderGrey,
  FullBattery,
  VeryHighRangeWithRoaming,
  HighRangeWithRoaming,
  MediumRangeWithRoaming,
  LowRangeWithRoaming,
  VeryLowRangeWithRoaming,
  ThinCheck,
  LongArrow,
  MarkAsRead,
  RecoveryBackup,
  RebootOs,
  FactoryReset,
  MuditaFlower,
  MuditaDarkLogo,
  ContactGoogle,
  Manage,
  MuditaLogoBg,
  Outlook,
  ContactFilled,
  Visible,
  Hidden,
  ThinFail,
  Warning,
  MarkAsUnread,
}
