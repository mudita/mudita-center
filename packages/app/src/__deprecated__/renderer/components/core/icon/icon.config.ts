/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { Image as ImageInterface } from "App/__deprecated__/renderer/interfaces/image.interface"
import Arrow from "App/__deprecated__/renderer/svg/arrow.svg"
import ArrowLongLeft from "App/__deprecated__/renderer/svg/arrow-long-left.svg"
import ArrowLongRight from "App/__deprecated__/renderer/svg/arrow-long-right.svg"
import AttachContact from "App/__deprecated__/renderer/svg/attach-contact.svg"
import BackupFolder from "App/__deprecated__/renderer/svg/backup-folder.svg"
import Battery from "App/__deprecated__/renderer/svg/battery.svg"
import BorderCheck from "App/__deprecated__/renderer/svg/border-check-icon.svg"
import ChargingBattery from "App/__deprecated__/renderer/svg/charging-battery.svg"
import Check from "App/__deprecated__/renderer/svg/check-icon.svg"
import CheckCircle from "App/__deprecated__/renderer/svg/check-circle.svg"
import CheckIndeterminate from "App/__deprecated__/renderer/svg/check-indeterminate.svg"
import Close from "App/__deprecated__/renderer/svg/close.svg"
import CloseWhite from "App/__deprecated__/renderer/svg/close-white.svg"
import Delete from "App/__deprecated__/renderer/svg/delete.svg"
import FilesManager from "App/__deprecated__/renderer/svg/files-manager.svg"
import HighBattery from "App/__deprecated__/renderer/svg/high-battery.svg"
import HighRange from "App/__deprecated__/renderer/svg/high-range.svg"
import LowBattery from "App/__deprecated__/renderer/svg/low-battery.svg"
import LowRange from "App/__deprecated__/renderer/svg/low-range.svg"
import Magnifier from "App/__deprecated__/renderer/svg/magnifier.svg"
import Calendar from "App/__deprecated__/renderer/svg/menu-calendar.svg"
import MediumBattery from "App/__deprecated__/renderer/svg/medium-battery.svg"
import MediumRange from "App/__deprecated__/renderer/svg/medium-range.svg"
import MenuFilesManager from "App/__deprecated__/renderer/svg/menu-files-manager.svg"
import MenuHelp from "App/__deprecated__/renderer/svg/menu-help.svg"
import MenuMeditation from "App/__deprecated__/renderer/svg/menu-meditation.svg"
import Message from "App/__deprecated__/renderer/svg/menu-messages.svg"
import MenuMusic from "App/__deprecated__/renderer/svg/menu-music.svg"
import MenuNews from "App/__deprecated__/renderer/svg/menu-news.svg"
import MenuOverview from "App/__deprecated__/renderer/svg/menu-overview.svg"
import MenuPhone from "App/__deprecated__/renderer/svg/menu-phone.svg"
import MenuSettings from "App/__deprecated__/renderer/svg/menu-settings.svg"
import MenuTethering from "App/__deprecated__/renderer/svg/menu-tethering.svg"
import MenuTools from "App/__deprecated__/renderer/svg/menu-tools.svg"
import ExternalLink from "App/__deprecated__/renderer/svg/external-link.svg"
import MuditaLogo from "App/__deprecated__/renderer/svg/mudita.svg"
import MuditaLogoWithText from "App/__deprecated__/renderer/svg/mudita_logo.svg"
import MuditaLogoVertical from "App/__deprecated__/renderer/svg/mudita-logo-vertical.svg"
import NoBattery from "App/__deprecated__/renderer/svg/no-battery.svg"
import NoRange from "App/__deprecated__/renderer/svg/no-range.svg"
import Refresh from "App/__deprecated__/renderer/svg/refresh.svg"
import Send from "App/__deprecated__/renderer/svg/send.svg"
import Sim from "App/__deprecated__/renderer/svg/sim.svg"
import Template from "App/__deprecated__/renderer/svg/template.svg"
import Upload from "App/__deprecated__/renderer/svg/upload.svg"
import UploadDark from "App/__deprecated__/renderer/svg/upload-dark.svg"
import Download from "App/__deprecated__/renderer/svg/download.svg"
import DownloadWhite from "App/__deprecated__/renderer/svg/download-white.svg"
import Calls from "App/__deprecated__/renderer/svg/phone.svg"
import Connection from "App/__deprecated__/renderer/svg/connection.svg"
import Contact from "App/__deprecated__/renderer/svg/contact.svg"
import NewContact from "App/__deprecated__/renderer/svg/new-contact.svg"
import Notes from "App/__deprecated__/renderer/svg/notes.svg"
import Notifications from "App/__deprecated__/renderer/svg/notifications.svg"
import Dial from "App/__deprecated__/renderer/svg/dial.svg"
import Playlist from "App/__deprecated__/renderer/svg/playlist.svg"
import PlusSign from "App/__deprecated__/renderer/svg/plus-sign.svg"
import Reload from "App/__deprecated__/renderer/svg/circle-arrow.svg"
import VoiceRecorder from "App/__deprecated__/renderer/svg/voice-recorder.svg"
import VeryHighBattery from "App/__deprecated__/renderer/svg/very-high-battery.svg"
import Pure from "App/__deprecated__/renderer/svg/pure.svg"
import Fail from "App/__deprecated__/renderer/svg/fail.svg"
import FailRed from "App/__deprecated__/renderer/svg/fail-red.svg"
import VeryHighRange from "App/__deprecated__/renderer/svg/very-high-range.svg"
import VeryLowBattery from "App/__deprecated__/renderer/svg/very-low-battery.svg"
import Info from "App/__deprecated__/renderer/svg/info.svg"
import More from "App/__deprecated__/renderer/svg/more.svg"
import Blocked from "App/__deprecated__/renderer/svg/blocked.svg"
import Forward from "App/__deprecated__/renderer/svg/forward.svg"
import Edit from "App/__deprecated__/renderer/svg/edit.svg"
import Favourites from "App/__deprecated__/renderer/svg/favourites.svg"
import Ice from "App/__deprecated__/renderer/svg/ice.svg"
import ArrowDown from "App/__deprecated__/renderer/svg/arrow-down.svg"
import ArrowDownWhite from "App/__deprecated__/renderer/svg/arrow-down-white.svg"
import DeleteBig from "App/__deprecated__/renderer/svg/delete-big.svg"
import Attachment from "App/__deprecated__/renderer/svg/attachment.svg"
import Remove from "App/__deprecated__/renderer/svg/remove.svg"
import SendButton from "App/__deprecated__/renderer/svg/send-button.svg"
import MenuContacts from "App/__deprecated__/renderer/svg/menu-contacts.svg"
import Backup from "App/__deprecated__/renderer/svg/backup.svg"
import Tooltip from "App/__deprecated__/renderer/svg/tooltip.svg"
import TetheringStatus from "App/__deprecated__/renderer/svg/tethering-status.svg"
import Templates from "App/__deprecated__/renderer/svg/templates.svg"
import MissedCall from "App/__deprecated__/renderer/svg/missed-call.svg"
import ConferenceCall from "App/__deprecated__/renderer/svg/conference-call.svg"
import IncomingCall from "App/__deprecated__/renderer/svg/incoming-call.svg"
import OutgoingCall from "App/__deprecated__/renderer/svg/outgoing-call.svg"
import AppleLogo from "App/__deprecated__/renderer/svg/apple.svg"
import GoogleLogo from "App/__deprecated__/renderer/svg/google.svg"
import SynchronizeContacts from "App/__deprecated__/renderer/svg/synchronize-contacts.svg"
import MusicGrey from "App/__deprecated__/renderer/svg/music-grey.svg"
import VoiceRecorderGrey from "App/__deprecated__/renderer/svg/voice-recorder-gray.svg"
import FullBattery from "App/__deprecated__/renderer/svg/full-battery.svg"
import VeryLowRange from "App/__deprecated__/renderer/svg/very-low-range.svg"
import VeryHighRangeWithRoaming from "App/__deprecated__/renderer/svg/very-high-range-roaming.svg"
import HighRangeWithRoaming from "App/__deprecated__/renderer/svg/high-range-roaming.svg"
import MediumRangeWithRoaming from "App/__deprecated__/renderer/svg/medium-range-roaming.svg"
import LowRangeWithRoaming from "App/__deprecated__/renderer/svg/low-range-roaming.svg"
import VeryLowRangeWithRoaming from "App/__deprecated__/renderer/svg/very-low-range-roaming.svg"
import CalendarIcon from "App/__deprecated__/renderer/svg/calendar.svg"
import ThinCheck from "App/__deprecated__/renderer/svg/thin-check.svg"
import LongArrow from "App/__deprecated__/renderer/svg/long-arrow.svg"
import MarkAsRead from "App/__deprecated__/renderer/svg/mark-as-read.svg"
import RecoveryBackup from "App/__deprecated__/renderer/svg/recovery-backup.svg"
import RebootOs from "App/__deprecated__/renderer/svg/reboot-os.svg"
import FactoryReset from "App/__deprecated__/renderer/svg/factory-reset.svg"
import MuditaFlower from "App/__deprecated__/renderer/svg/mudita-flower.svg"
import MuditaDarkLogo from "App/__deprecated__/renderer/svg/mudita-dark-logo.svg"
import ContactGoogle from "App/__deprecated__/renderer/svg/contact-google.svg"
import Manage from "App/__deprecated__/renderer/svg/manage.svg"
import MuditaLogoBg from "App/__deprecated__/renderer/svg/muditalogo-bg.svg"
import Outlook from "App/__deprecated__/renderer/svg/outlook.svg"
import ContactFilled from "App/__deprecated__/renderer/svg/contact-filled.svg"
import ThinFail from "App/__deprecated__/renderer/svg/thin-fail.svg"
import Support from "App/__deprecated__/renderer/svg/support.svg"
import Visible from "App/__deprecated__/renderer/svg/password-show.svg"
import Hidden from "App/__deprecated__/renderer/svg/password-hide.svg"
import Cloud from "App/__deprecated__/renderer/svg/cloud.svg"
import Warning from "App/__deprecated__/renderer/svg/warning.svg"
import MarkAsUnread from "App/__deprecated__/renderer/svg/mark-as-unread.svg"
import { FunctionComponent } from "App/__deprecated__/renderer/types/function-component.interface"
import { IconType } from "App/__deprecated__/renderer/components/core/icon/icon-type"

const typeToIcon: Partial<Record<IconType, typeof Arrow>> = {
  [IconType.Arrow]: Arrow,
  [IconType.ArrowLongLeft]: ArrowLongLeft,
  [IconType.ArrowLongRight]: ArrowLongRight,
  [IconType.AttachContact]: AttachContact,
  [IconType.BackupFolder]: BackupFolder,
  [IconType.Battery]: Battery,
  [IconType.BorderCheckIcon]: BorderCheck,
  [IconType.Calendar]: Calendar,
  [IconType.CalendarIcon]: CalendarIcon,
  [IconType.ChargingBattery]: ChargingBattery,
  [IconType.Check]: Check,
  [IconType.CheckCircle]: CheckCircle,
  [IconType.CheckIndeterminate]: CheckIndeterminate,
  [IconType.Close]: Close,
  [IconType.CloseWhite]: CloseWhite,
  [IconType.Cloud]: Cloud,
  [IconType.Delete]: Delete,
  [IconType.FilesManager]: FilesManager,
  [IconType.HighBattery]: HighBattery,
  [IconType.HighRange]: HighRange,
  [IconType.LowBattery]: LowBattery,
  [IconType.LowRange]: LowRange,
  [IconType.Magnifier]: Magnifier,
  [IconType.MediumBattery]: MediumBattery,
  [IconType.MenuContacts]: MenuContacts,
  [IconType.MediumRange]: MediumRange,
  [IconType.MenuFilesManager]: MenuFilesManager,
  [IconType.MenuHelp]: MenuHelp,
  [IconType.MenuMeditation]: MenuMeditation,
  [IconType.MenuMusic]: MenuMusic,
  [IconType.MenuNews]: MenuNews,
  [IconType.MenuOverview]: MenuOverview,
  [IconType.MenuPhone]: MenuPhone,
  [IconType.MenuSettings]: MenuSettings,
  [IconType.MenuTethering]: MenuTethering,
  [IconType.MenuTools]: MenuTools,
  [IconType.Message]: Message,
  [IconType.ExternalLink]: ExternalLink,
  [IconType.MuditaLogo]: MuditaLogo,
  [IconType.MuditaLogoWithText]: MuditaLogoWithText,
  [IconType.MuditaLogoVertical]: MuditaLogoVertical,
  [IconType.NoBattery]: NoBattery,
  [IconType.NoRange]: NoRange,
  [IconType.Refresh]: Refresh,
  [IconType.Send]: Send,
  [IconType.Sim]: Sim,
  [IconType.Support]: Support,
  [IconType.Template]: Template,
  [IconType.Tooltip]: Tooltip,
  [IconType.TetheringStatus]: TetheringStatus,
  [IconType.Templates]: Templates,
  [IconType.Calls]: Calls,
  [IconType.Connection]: Connection,
  [IconType.Contact]: Contact,
  [IconType.NewContact]: NewContact,
  [IconType.Dial]: Dial,
  [IconType.Notes]: Notes,
  [IconType.Notifications]: Notifications,
  [IconType.Playlist]: Playlist,
  [IconType.PlusSign]: PlusSign,
  [IconType.Reload]: Reload,
  [IconType.Upload]: Upload,
  [IconType.UploadDark]: UploadDark,
  [IconType.VeryHighBattery]: VeryHighBattery,
  [IconType.VeryHighRange]: VeryHighRange,
  [IconType.VeryLowBattery]: VeryLowBattery,
  [IconType.VoiceRecorder]: VoiceRecorder,
  [IconType.Info]: Info,
  [IconType.Download]: Download,
  [IconType.DownloadWhite]: DownloadWhite,
  [IconType.Pure]: Pure,
  [IconType.Fail]: Fail,
  [IconType.FailRed]: FailRed,
  [IconType.More]: More,
  [IconType.Blocked]: Blocked,
  [IconType.Forward]: Forward,
  [IconType.Edit]: Edit,
  [IconType.Favourites]: Favourites,
  [IconType.Ice]: Ice,
  [IconType.ArrowDown]: ArrowDown,
  [IconType.ArrowDownWhite]: ArrowDownWhite,
  [IconType.DeleteBig]: DeleteBig,
  [IconType.Attachment]: Attachment,
  [IconType.Remove]: Remove,
  [IconType.SendButton]: SendButton,
  [IconType.Backup]: Backup,
  [IconType.Apple]: AppleLogo,
  [IconType.Google]: GoogleLogo,
  [IconType.SynchronizeContacts]: SynchronizeContacts,
  [IconType.MissedCall]: MissedCall,
  [IconType.ConferenceCall]: ConferenceCall,
  [IconType.IncomingCall]: IncomingCall,
  [IconType.OutgoingCall]: OutgoingCall,
  [IconType.MusicGrey]: MusicGrey,
  [IconType.VoiceRecorderGrey]: VoiceRecorderGrey,
  [IconType.FullBattery]: FullBattery,
  [IconType.VeryLowRange]: VeryLowRange,
  [IconType.VeryHighRangeWithRoaming]: VeryHighRangeWithRoaming,
  [IconType.HighRangeWithRoaming]: HighRangeWithRoaming,
  [IconType.MediumRangeWithRoaming]: MediumRangeWithRoaming,
  [IconType.LowRangeWithRoaming]: LowRangeWithRoaming,
  [IconType.VeryLowRangeWithRoaming]: VeryLowRangeWithRoaming,
  [IconType.MarkAsRead]: MarkAsRead,
  [IconType.ThinCheck]: ThinCheck,
  [IconType.MuditaFlower]: MuditaFlower,
  [IconType.LongArrow]: LongArrow,
  [IconType.RecoveryBackup]: RecoveryBackup,
  [IconType.RebootOs]: RebootOs,
  [IconType.FactoryReset]: FactoryReset,
  [IconType.MuditaDarkLogo]: MuditaDarkLogo,
  [IconType.ContactGoogle]: ContactGoogle,
  [IconType.MuditaLogoBg]: MuditaLogoBg,
  [IconType.Manage]: Manage,
  [IconType.Outlook]: Outlook,
  [IconType.ContactFilled]: ContactFilled,
  [IconType.Visible]: Visible,
  [IconType.Hidden]: Hidden,
  [IconType.ThinFail]: ThinFail,
  [IconType.Warning]: Warning,
  [IconType.MarkAsUnread]: MarkAsUnread,
}

export const getIconType = (
  icon: IconType = IconType.Message
): FunctionComponent<ImageInterface> => typeToIcon[icon] || Message

export function getEnumName(type: IconType): string
export function getEnumName(type?: IconType): null
export function getEnumName(type?: IconType): string | null {
  return type !== undefined ? IconType[type] : null
}
