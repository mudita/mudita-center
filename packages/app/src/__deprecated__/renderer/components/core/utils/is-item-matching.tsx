/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

export type IsItemMatching = (item: any, searchString: string) => boolean
export const isItemValueMatching: IsItemMatching = (
  itemValue: string,
  search
) => itemValue.toLowerCase().includes(search.toLowerCase())
