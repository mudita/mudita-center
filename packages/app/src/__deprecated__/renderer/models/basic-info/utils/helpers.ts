/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { SimCard } from "App/__deprecated__/renderer/models/basic-info/basic-info.typings"

export const getActiveNetworkFromSim = (simCards: SimCard[] = []) => {
  const activeSimCard = simCards.filter((simCard) => simCard.active)
  if (activeSimCard.length === 0) {
    return "No connection"
  }
  return activeSimCard[0].network
}
export const getActiveNetworkLevelFromSim = (simCards: SimCard[] = []) => {
  const activeSimCard = simCards.filter((simCard) => simCard.active)
  if (activeSimCard.length === 0) {
    return "No connection"
  }
  return activeSimCard[0].networkLevel
}
