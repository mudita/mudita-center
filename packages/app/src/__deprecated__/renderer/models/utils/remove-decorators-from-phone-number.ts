/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

export const removeDecoratorsFromPhoneNumber = (phoneNumber = ""): string => {
  const rawPhoneNumber = phoneNumber.replace(/[^\d]/g, "")
  if (rawPhoneNumber === "") {
    return phoneNumber.replace(/\s/g, "").toLowerCase()
  } else {
    return rawPhoneNumber
  }
}
