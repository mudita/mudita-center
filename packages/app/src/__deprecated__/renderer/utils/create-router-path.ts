/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { URL_MAIN, URL_TABS } from "App/__deprecated__/renderer/constants/urls"

//if params contains sensitive data, they should be added to scrubFields array in logger.ts
const createRouterPath = (
  link:
    | typeof URL_MAIN[keyof typeof URL_MAIN]
    | typeof URL_TABS[keyof typeof URL_TABS],
  params: Record<string, string>
): string => {
  return `${link}?${new URLSearchParams(params).toString()}`
}

export default createRouterPath
