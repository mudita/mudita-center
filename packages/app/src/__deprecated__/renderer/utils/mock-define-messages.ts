/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { defineMessages } from "react-intl"

export const mockDefineMessages = (id = "module.news") => {
  const messages = defineMessages({
    exampleMessage: { id },
  })
  return messages.exampleMessage
}
