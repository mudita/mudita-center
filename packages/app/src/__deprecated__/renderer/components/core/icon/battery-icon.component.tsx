/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { FunctionComponent } from "App/__deprecated__/renderer/types/function-component.interface"
import * as React from "react"
import Icon, {
  Props as IconProps,
} from "App/__deprecated__/renderer/components/core/icon/icon.component"
import { IconType } from "App/__deprecated__/renderer/components/core/icon/icon-type"

export interface BatteryIconProps {
  level: number
  charging?: boolean
}

const getInteractiveBatteryIcon = (
  batteryLevel: number,
  charging: boolean,
  size: number,
  rest: Partial<IconProps>
) => {
  switch (true) {
    case charging:
      return <Icon type={IconType.ChargingBattery} {...rest} />
    case batteryLevel === 1:
      return <Icon type={IconType.FullBattery} {...rest} />
    case batteryLevel > 0.8:
      return <Icon type={IconType.VeryHighBattery} {...rest} />
    case batteryLevel > 0.6:
      return <Icon type={IconType.HighBattery} {...rest} />
    case batteryLevel > 0.4:
      return <Icon type={IconType.MediumBattery} {...rest} />
    case batteryLevel > 0.2:
      return <Icon type={IconType.LowBattery} {...rest} />
    case batteryLevel > 0:
      return <Icon type={IconType.VeryLowBattery} {...rest} />
    default:
      return <Icon type={IconType.NoBattery} {...rest} />
  }
}

const BatteryIcon: FunctionComponent<BatteryIconProps & IconProps> = ({
  level,
  charging = false,
  height = 2,
  ...rest
}) => {
  return <>{getInteractiveBatteryIcon(level, charging, height, rest)}</>
}

export default BatteryIcon
