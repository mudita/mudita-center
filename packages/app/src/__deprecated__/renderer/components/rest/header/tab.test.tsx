/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import "@testing-library/jest-dom"
import React from "react"
import { MemoryRouter } from "react-router-dom"
import Tab from "App/__deprecated__/renderer/components/rest/header/tab.component"
import { mockDefineMessages } from "App/__deprecated__/renderer/utils/mock-define-messages"
import { renderWithThemeAndIntl } from "App/__deprecated__/renderer/utils/render-with-theme-and-intl"
import { IconType } from "App/__deprecated__/renderer/components/core/icon/icon-type"

const message = mockDefineMessages("module.phone")
const currentLocation = "/phone"

test("matches snapshot", () => {
  const { container } = renderWithThemeAndIntl(
    <MemoryRouter initialEntries={[currentLocation]}>
      <Tab icon={IconType.Check} label={message} url={currentLocation} />
    </MemoryRouter>
  )
  expect(container.firstChild).toMatchSnapshot()
})

test("has correct text content", () => {
  const { container } = renderWithThemeAndIntl(
    <MemoryRouter initialEntries={[currentLocation]}>
      <Tab icon={IconType.Check} label={message} />
    </MemoryRouter>
  )
  const tabNode = container.firstChild
  expect(tabNode).toHaveTextContent("module.phone")
})
