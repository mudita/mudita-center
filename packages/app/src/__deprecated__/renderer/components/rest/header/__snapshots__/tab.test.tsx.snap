// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
.c3 {
  margin: 0;
  font-size: 1.2rem;
  line-height: 1.67;
  font-weight: 400;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  text-transform: uppercase;
  color: #000000;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 2.4rem;
  width: 2.4rem;
  position: relative;
  font-style: normal;
}

.c1 svg {
  width: 100%;
  height: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  padding: 0.1rem 0.7rem;
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  overflow: hidden;
  -webkit-transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  width: 17.6rem;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  height: 100%;
  padding: 0.8rem 0;
  border: none;
  position: relative;
  width: 100%;
}

.c0:after {
  content: "";
  position: absolute;
  display: block;
  bottom: 0;
  left: 0;
  width: 0;
  height: 0.2rem;
  background-color: #000000;
  -webkit-transition: width 500ms ease;
  transition: width 500ms ease;
}

.c0 p {
  color: #6a6a6a;
}

.c0 svg {
  opacity: 0.75;
}

.c0:hover p {
  color: #3b3f42;
}

.c0:hover g path {
  fill: #3b3f42;
}

.c0.active p {
  color: #000000;
}

.c0.active g path {
  fill: #000000;
}

.c0.active:after {
  width: 100%;
}

.c2 {
  margin: 0 0.8rem 0 0;
}

.c4 {
  white-space: nowrap;
}

<a
  aria-current="page"
  class="c0 active"
  displaystyle="8"
  href="/phone"
  size="1"
>
  <span
    class="c1 c2"
    data-testid="icon-Check"
    height="2.4"
    width="2.4"
  >
    <test-file-stub
      image="test-file-stub"
    />
  </span>
  <p
    class="c3 c4"
    color="primary"
  >
    [value] module.phone
  </p>
</a>
`;
